### ISSUE TITLE
[BUG]Reparation of a big string dask_cudf dataframe fails on the  current nightly

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#3859]
**Describe the bug**

Reparation of a big string dask_cudf dataframe is failing on the nightly currently. For context below used to work on `6ths nightly`.

The peak memory on **6ths nightly** is as follows:

```python
### before repartition
GPU #0: 4764 MB/32480 MB [Peak mem: 4764 MB]
GPU #1: 3672 MB/32480 MB [Peak mem: 3672 MB]
### after repartition
GPU #0: 1634 MB/32480 MB [Peak mem: 4842 MB]
GPU #1: 6134 MB/32480 MB [Peak mem: 12024 MB] 
```

On  `0.12.0b200120` i get below:
```python
#Before Repartition
GPU #0: 6627 MB/32480 MB [Peak mem: 6627 MB]
GPU #1: 6036 MB/32480 MB [Peak mem: 6036 MB]

#After Repartition
GPU #0: 11603 MB/32480 MB [Peak mem: 11603 MB]
GPU #1: 1708 MB/32480 MB [Peak mem: 17670 MB]
```

**Steps/Code to reproduce bug**
```python
from dask_cuda import LocalCUDACluster
from dask.distributed import Client,wait
import dask_cudf
import cudf
cluster = LocalCUDACluster(CUDA_VISIBLE_DEVICES="0,1")
client = Client(cluster)
final = dask_cudf.read_parquet('/datasets/vjawa/test_data/*.parquet')
final = final.persist()
_ = wait(final)
print("persisted = {}".format(len(final)))
final = final.repartition(npartitions=1).persist()
_ = wait(final)
print("Length after repartition = {}".format(len(final)))
```

**Expected behavior**

I would expect it to work without memory issues. 

**Additional context**
There has been some discussion about it on our internal slack .

See link: https://nvidia.slack.com/archives/CGLPWR4HJ/p1579641561076800

**Comment by kkraus14** [comment]
Note: this is throwing a `cudaErrorIllegalAddress: an illegal memory access was encountered` so it's not an OOM error

**Comment by kkraus14** [comment]
Fixed by #3866
